numpy/_core/_add_newdocs.py

```diff
@@ -6813,3 +6813,56 @@ def refer_to_array_attribute(attr, method=True):
     Abstract base class of all character string scalar types.
 
     """)
+
+add_newdoc('numpy._core.multiarray', 'StringDType',
+    """
+    StringDType(/, size, *, na_object=np._NoValue, coerce=True)
+
+    Create a StringDType instance.
+
+    StringDType can be used to store UTF-8 encoded variable-width strings in
+    a NumPy array.
+
+    Parameters
+    ----------
+    size: integer
```

Comment:
This was needed when stringdtype lived outside numpy so that we could write code that would work with both the fixed-width and variable-width string dtypes. Like [here](https://github.com/numpy/numpy/blob/0a4b2b83eaf3479f352a7fe5a4b378a169ab48f0/numpy/_core/defchararray.py#L441).

If stringdtype is inside numpy we could just add paths in all the implementations that use `_vec_string` that don't figure out what the output size is supposed to be and instead just pass the input dtype directly.

Would you be OK if I do that in a followup and remove this argument then?

Reply:
Yes, I mainly want the documentation to be correct when this goes in!